app/controllers/root_controller.rb

```diff
@@ -109,12 +110,58 @@ def publication
       end
 
       @interaction_details = prepare_interaction_details(@publication, authority_slug, snac)
+    elsif @publication.format == 'local_transaction'
+      @location = fetch_location @postcode
+
+      if @location.nil?
+        @postcode_error = "fullPostcodeNoMapitMatch"
```

Comment:
When `fetch_location` returns `nil`, it's the equivalent of a 404 from Mapit. That case happens when there is no match but it is a Postcode.